core/unit_test/TestExecutionEnvironmentNonInitializedOrFinalized.cpp

```diff
@@ -183,4 +183,258 @@ TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
       "kokkos_free\\(\\) \\*\\*after\\*\\* Kokkos::finalize\\(\\) was called");
 }
 
+template <typename Type>
+class EmptyReduceFunctor {
+ public:
+  using size_type = typename Kokkos::DefaultExecutionSpace::size_type;
+  KOKKOS_INLINE_FUNCTION
+  void join(Type&, const Type&) const {}
+  KOKKOS_INLINE_FUNCTION
+  void operator()(size_type, Type&) const {}
+  KOKKOS_INLINE_FUNCTION
+  void final(Type&) const {}
+};
+
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
+       parallel_for_1) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  EXPECT_DEATH(
+      {
+        Kokkos::initialize();
+        Kokkos::RangePolicy<> policy(0, 0);
+        Kokkos::finalize();
+
+        Kokkos::parallel_for("parallel_for_1", policy, KOKKOS_LAMBDA(int){});
+      },
+      "Kokkos ERROR: attempting to call "
+      "parallel_for\\(\\) \\*\\*after\\*\\* Kokkos::finalize\\(\\) was called. "
+      "Concerns parallel_for_1 with exec policy RangePolicy.");
+}
+
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
+       parallel_for_2) {
```

Comment:
"parallel_for_1" or "parallel_for_2" os not descriptive.
I don't quite get why we would bother separating them out so maybe we can just have a single `parallel_for`.

Reply:
when looking at the log you one would not be able to tell what 1 and 2 are.
I am saying maybe we don't need to separate them out so we don't have to come up with decent names.